tools/permissions.ts

```diff
@@ -0,0 +1,298 @@
+    case 'build':
+      return BuildResourceSchema.parse(response);
+  }
+}
+
+export async function replacePermissionResource(
+  name: string, 
+  resourceType: 'artifact' | 'release_bundle' | 'build',
+  resource: z.infer<typeof ArtifactResourceSchema> | z.infer<typeof ReleaseBundleResourceSchema> | z.infer<typeof BuildResourceSchema>
+) {
+  const response = await jfrogRequest(`/access/api/v2/permissions/${name}/${resourceType}`, {
+    method: "PUT",
+    body: resource
+  });
+  switch (resourceType) {
+    case 'artifact':
+      return ArtifactResourceSchema.parse(response);
+    case 'release_bundle':
+      return ReleaseBundleResourceSchema.parse(response);
+    case 'build':
+      return BuildResourceSchema.parse(response);
+  }
+}
+
+export async function deletePermissionResource(name: string, resourceType: 'artifact' | 'release_bundle' | 'build') {
+  await jfrogRequest(`/access/api/v2/permissions/${name}/${resourceType}`, {
+    method: "DELETE",
+  });
+  return { success: true };
+}
+
+/* Tools Section */
+
+const listPermissionTargetsTool = {
+  name: "jfrog_list_permission_targets",
+  description: "Get a list of all permission targets in the JFrog platform",
+  inputSchema: zodToJsonSchema(z.object({
+    cursor: z.string().optional().describe("Cursor for pagination"),
+    limit: z.number().optional().describe("Limit the number of results")
+  })),
+  outputSchema: zodToJsonSchema(PermissionListSchema),
+  handler: async (args: any) => {
+    return await getAllPermissionTargets(args.cursor, args.limit);
+  }
+};
+
+const getPermissionTargetTool = {
+  name: "jfrog_get_permission_target",
+  description: "Get detailed information about a specific permission target",
+  inputSchema: zodToJsonSchema(z.object({
+    name: z.string().describe("The name of the permission target to retrieve")
+  })),
+  outputSchema: zodToJsonSchema(PermissionTargetSchema),
+  handler: async (args: any) => {
+    return await getPermissionTarget(args.name);
+  }
+};
+
+const createPermissionTargetTool = {
+  name: "jfrog_create_permission_target",
+  description: "Create a new permission target in the JFrog platform",
+  inputSchema: zodToJsonSchema(PermissionTargetSchema),
+  outputSchema: zodToJsonSchema(PermissionTargetSchema),
+  handler: async (args: any) => {
+    return await createPermissionTarget(args);
+  }
+};
+
+const updatePermissionTargetTool = {
+  name: "jfrog_update_permission_target",
+  description: "Update an existing permission target in the JFrog platform",
+  inputSchema: zodToJsonSchema(z.object({
+    name: z.string().describe("The name of the permission target to update"),
+    target: PermissionTargetSchema.omit({ name: true })
+  })),
+  outputSchema: zodToJsonSchema(PermissionTargetSchema),
+  handler: async (args: any) => {
+    const { name, target } = args;
+    return await updatePermissionTarget(name, { name, ...target });
+  }
+};
+
+const deletePermissionTargetTool = {
```

Comment:
Since delete operations can be risky, I suggest we hold off on including them for now. Let's comment this out until we gain more confidence in the tools and their behavior